server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -574,6 +574,7 @@ protected Node(
 
             runnableTaskListener = new AtomicReference<>();
             final ThreadPool threadPool = new ThreadPool(settings, runnableTaskListener, executorBuilders.toArray(new ExecutorBuilder[0]));
+            pluginsService.initializePlugins(threadPool);
```

Comment:
The initialization logic belongs to `IdentityService` I believe

Reply:
Moved initialization to the IdentityService and created a special initialization method for IdentityAwarePlugins.